src/rest_api/classes/phenotype/core.clj

```diff
@@ -3,46 +3,266 @@
+
+    :Haplo_insufficient
+    (if
+      (contains? holder :phenotype-info/haplo-insufficient)
+      (obj/tag-obj
+        (obj/humanize-ident
+          (:phenotype-info.paternal/value
+            (:phenotype-info/haplo-insufficient holder)))))
+
+    :Variation_effect
+    (if (contains? holder :phenotype-info/variation-effect)
+      (first
+       ;; we should actually display all of them but catalyst template
+       ;; not displaying nested array
+       (for [ve (:phenotype-info/variation-effect holder)]
+         (remove
+          nil?
+          [(obj/tag-obj
+            (obj/humanize-ident
+             (:phenotype-info.variation-effect/value ve)))
+           (if
+               (contains? ve :evidence/person-evidence)
+             (obj/tag-obj "Person_evidence"))
+           (if
+               (contains? ve :evidence/curator-confirmed)
+             (obj/tag-obj "Curator_confirmed"))
+           (if
+               (contains? ve :evidence/paper-evidence)
+             (obj/tag-obj "Paper_evidence"))]))))
+
+    :Affected_by_molecule
+    (if
+      (contains? holder :phenotype-info/molecule)
+      (for [m (:phenotype-info/molecule holder)]
+        (pack-obj (:phenotype-info.molecule/molecule m))))
+
+    :Affected_by_pathogen
+    (if
+      (contains? holder :phenotype-info/pathogen)
+      (for [m (:phenotype-info/pathogen holder)]
+        (pack-obj (:phenotype-info.molecule/species m))))
+
+    :Ease_of_scoring
+    (if
+      (contains? holder :phenotype-info/ease-of-scoring)
+      (obj/tag-obj
+        (obj/humanize-ident
+          (:phenotype-info.ease-of-scoring/value
+            (:phenotype-info/ease-of-scoring holder)))))
+
+    :keys (keys holder)
+
+    :Caused_by_gene
+    (if
+      (contains? holder :phenotype-info/caused-by-gene)
+      (let [cbgs (:phenotype-info/caused-by-gene holder)]
+        (for [cbg cbgs]
+          (pack-obj (:phenotype-info.caused-by-gene/gene cbg)))))
+
+    :Phenotype_assay
+    (if
+      (contains? holder :phenotype-info/strain)
+      (obj/tag-obj "Strain"))
+
+    :Male_mating_efficiency
+    (if
+      (contains? variation :variation/male-mating-efficiency)
+      (obj/humanize-ident
+        (:variation.male-mating-efficiency/value
+          (:variation/male-mating-efficiency variation))))
+
+    :Temperature_sensitive
+    (if
+      (or
+        (contains? holder :phenotype-info/heat-sensitive)
+        (contains? holder :phenotype-info/cold-sensitive))
+        (conj
+          (if (contains? holder :phenotype-info/heat-sensitive)
+            (obj/tag-obj "Heat-sensitive"))
+          (if (contains? holder :phenotype-info/cold-sensitive)
+            (obj/tag-obj "Cold-sensitive"))))
+
+    :Treatment
+    (if
+      (contains? holder :phenotype-info/treatment)
+      (first (for [treatment-holder (:phenotype-info/treatment holder)
+        :let [text (:phenotype-info.treatment/text treatment-holder)]]
+        (if (= text "") nil text))))
```

Comment:
Repetion of this pattern a lot:
(if (= text "") nil text)

Perhaps use the pattern:
(if-not (str/blank? text)
   text))